odoo/addons/base/ir/ir_sequence.py

```diff
@@ -144,14 +144,17 @@ def create(self, values):
         """ Create a sequence, in implementation == standard a fast gaps-allowed PostgreSQL sequence is used.
         """
         seq = super(IrSequence, self).create(values)
+        self.clear_caches()
```

Comment:
That this has the desired effect depends on the behaviour of `Registry.reset_changes()`; see [this bug report](https://github.com/odoo/odoo/issues/35289). In summary, clearing caches here does not protect a future call to `seq_by_code` from caching the created id which could then disappear due to a rollback. However, under normal conditions a rollback will be triggered by an exception, which will fire `Registry.reset_changes()` which will see that caches have been cleared and *re-clear* them.

Reply:
Actually, does create() need to clear the cache at all?  There is no way that seq_by_code could plausibly have cached a result for a sequence that had not yet been created.

Reply:
Yep, without this the first of the new tests fails. Adding it makes the test pass - and fixes the observed issue that arose in our addon. What happens is a new sequence is created, `seq_by_code` caches the result, then the transaction is rolled back. Putting clear here *does* still allow that id into the cache, but the behaviour of `reset_changes()` is such that it gets cleared *again* (if an exception is encountered, triggering a rollback).

`clear_caches` does double duty as both clearing caches and signalling that data that can be cached has been modified, and thus that the caches may need to be cleared again after a rollback.

An option that breaks encapsulation (arguably just in a different way) but makes more sense would be to change this line to `self.pool.cache_invalidated = True`. This is the flag that `clear_caches` sets and which is checked for in `reset_changes`.